addons/mail/static/src/owl/components/composer/composer.js

```diff
@@ -341,6 +350,17 @@ class Composer extends Component {
         this._postMessage();
     }
 
+    _setTextInputHtmlContent(textInputContent) {
+        //Removing unwanted extra spaces from message
+        let value = _.escape(textInputContent).trim();
+        value = value.replace(/(\r|\n){2,}/g, '<br/><br/>');
+        value = value.replace(/(\r|\n)/g, '<br/>');
+
+        // prevent html space collapsing
+        value = value.replace(/ /g, '&nbsp;').replace(/([^>])&nbsp;([^<])/g, '$1 $2');
```

Comment:
I feel like this is a bad idea to use non breaking spaces for every single space... Those exist and are different for a reason.

I feel like only "multiple" spaces should be replaced, and that is still a questionable spec (html has `pre` tags to avoid those issues when necessary).

Also it's even more strange to do this before posting the message. I would feel at best it should only be done at display, but stored as-is in database.

Reply:
We'll see that later